src/init/shared/reload-page-on-consent-change.ts

```diff
@@ -0,0 +1,26 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { once } from 'lodash-es';
+
+let initialConsentState: ConsentState | undefined;
+
+/**
+ * If consent has been set, and if consent then changes, reload the page so the correct
+ * state is reflected in the rendered page
+ */
+const _reloadPageOnConsentChange = (): void => {
+	onConsentChange((consent) => {
+		if (initialConsentState === undefined) {
+			initialConsentState = consent;
+		}
+		if (initialConsentState.canTarget !== consent.canTarget) {
+			window.location.reload();
+		}
+	}, true);
+};
+
+const reloadPageOnConsentChange = once(() =>
```

Comment:
Not sure why we need to once() as I thought we would only call this in the boot script?

Reply:
Good spot - I think this is from the original PR back in frontend. I've removed it and seems to work fine without!